src/query/service/src/sessions/query_ctx.rs

```diff
@@ -592,6 +585,31 @@ impl QueryContext {
     pub fn clear_table_meta_timestamps_cache(&self) {
         self.shared.table_meta_timestamps.lock().clear();
     }
+
+    pub fn get_materialized_cte_senders(
+        &self,
+        cte_name: &str,
+        cte_ref_count: usize,
+    ) -> Vec<Sender<DataBlock>> {
+        let mut senders = vec![];
+        let mut receivers = vec![];
+        for _ in 0..cte_ref_count {
+            let (sender, receiver) = async_channel::unbounded();
```

Comment:
This causes all data to be stored entirely in memory. Will future PRs support CTE spilling?